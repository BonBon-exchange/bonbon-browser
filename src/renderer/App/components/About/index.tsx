import { About } from './About';

export * from './About';
export default About;
