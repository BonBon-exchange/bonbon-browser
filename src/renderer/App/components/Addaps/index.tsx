import { Addaps } from './Addaps';

export * from './Addaps';
export default Addaps;
