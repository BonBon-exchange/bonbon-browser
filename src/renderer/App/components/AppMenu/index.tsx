import { AppMenu } from './AppMenu';

export * from './AppMenu';
export default AppMenu;
