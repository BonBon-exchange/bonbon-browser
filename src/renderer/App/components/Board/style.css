.Board__container {
  display: flex;
  height: 300vh;
  width: calc(100% - 50px);
  top: 0px;
  left: 50px;
  position: absolute;
  background: linear-gradient(
    200.96deg,
    #fedc2a -29.09%,
    #dd5789 51.77%,
    #7a2c9e 129.35%
  );
}
