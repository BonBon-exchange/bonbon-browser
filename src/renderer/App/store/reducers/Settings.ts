import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { flatten } from 'flat';

export type SettingsState = {
  settings: {
    [key: string]: any;
  };
};

export const initialState: SettingsState = {
  settings: {},
};

// The settingsSlice for managing application settings
export const settingsSlice = createSlice({
  name: 'settings',
  initialState,
  reducers: {
    // Set a single setting value
    setSetting: (state, action: PayloadAction<{ key: string; value: any }>) => {
      state.settings[action.payload.key] = action.payload.value;
      window.app.config.set({
        key: action.payload.key,
        value: action.payload.value,
      });
    },

    // Sync settings by replacing the whole state with new settings
    syncSettings: (state, action: PayloadAction<Record<string, any>>) => {
      state.settings = { ...state.settings, ...flatten(action.payload) }; // Merge new settings with existing state
    },
  },
});

// Exporting the actions
export const { setSetting, syncSettings } = settingsSlice.actions;

// Exporting the reducer
export default settingsSlice.reducer;
