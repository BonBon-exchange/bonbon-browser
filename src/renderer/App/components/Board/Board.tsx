/* eslint-disable promise/always-return */
/* eslint-disable @typescript-eslint/ban-ts-comment */
/* eslint-disable react/jsx-props-no-spreading */
/* eslint-disable import/prefer-default-export */
/* eslint-disable no-use-before-define */
import clsx from 'clsx';
import { AnimatePresence } from 'framer-motion';
import React, { useCallback, useEffect, useState } from 'react';

import { Browser } from 'renderer/App/components/Browser';
import { BrowserProps } from 'renderer/App/components/Browser/Types';
import { useBoard } from 'renderer/App/hooks/useBoard';
import { useBrowserMethods } from 'renderer/App/hooks/useBrowserMethods';
import { useStoreHelpers } from 'renderer/App/hooks/useStoreHelpers';
import { useAppDispatch } from 'renderer/App/store/hooks';
import {
  setActiveBrowser,
  setBoardHeight,
} from 'renderer/App/store/reducers/Board';

import { BoardProps } from './Types';

import './style.scss';

export const Board: React.FC<BoardProps> = ({ isFullSize, boardId }) => {
  const board = useBoard();
  const dispatch = useAppDispatch();
  const { focus } = useBrowserMethods();
  const [items, setItems] = useState<BrowserProps[]>([]);
  const helpers = useStoreHelpers();

  const boardContainer = document.querySelector('#Board__container');

  const makeBrowsers = useCallback((sorted: BrowserProps[]) => {
    return sorted.map((b) => <Browser {...b} key={b.id} firstRendering />);
  }, []);

  const contextMenuListener = (e: MouseEvent) => {
    e.preventDefault();
    window.app.tools.showBoardContextMenu({ x: e.clientX, y: e.clientY });
  };

  useEffect(() => {
    if (!board.isFullSize) {
      setTimeout(() => {
        helpers.board
          .distributeWindowsByOrder(board.browsers)
          .then(() => {
            setTimeout(
              () => board.activeBrowser && focus(board.activeBrowser),
              300
            );
          })
          .catch(console.log);
      }, 300);
    }
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [board.isFullSize]);

  useEffect(() => {
    const toSort = [...board.browsers];
    const sorted = toSort.sort((a, b) => {
      return a.id > b.id ? 1 : -1;
    });
    setItems(sorted);

    if (boardContainer) {
      const max = board.browsers.reduce(
        (acc, val) => Math.max(acc, val.top + val.height + 100),
        0
      );

      // @ts-ignore
      boardContainer.style.height = `${Number(max)}px`;

      dispatch(setBoardHeight(max));
    }
  }, [board.browsers, dispatch, boardContainer]);

  useEffect(() => {
    window.app.board.setWindowsCount({
      boardId: board.id,
      count: board.browsers.length,
    });
  }, [board.browsers.length, board.id]);

  useEffect(() => {
    if (board.activeBrowser) {
      focus(board.activeBrowser, true);
      dispatch(setActiveBrowser(board.activeBrowser));
    }
  }, [board.activeBrowser, focus, dispatch]);

  // focus activeBrowser when componentDidMount
  useEffect(() => {
    setTimeout(() => {
      if (board.activeBrowser) focus(board.activeBrowser);
      if (boardId) {
        window.app.browser
          .getUrlToOpen()
          .then((res) => {
            if (res) helpers.browser.add({ url: res });
          })
          .catch(console.log);
      }
    }, 0);
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  useEffect(() => {
    document
      .getElementById('Board__container')
      ?.addEventListener('contextmenu', contextMenuListener);

    return () =>
      document
        .getElementById('Board__container')
        ?.removeEventListener('contextmenu', contextMenuListener);
  }, []);

  return (
    <div
      id="Board__container"
      className={clsx({
        'Board__is-maximized': board.isFullSize || isFullSize,
        'Board__isnt-maximized': !board.isFullSize && !isFullSize,
      })}
    >
      <AnimatePresence>{makeBrowsers(items)}</AnimatePresence>
      <div className="Board__edge-snap-left" />
      <div className="Board__edge-snap-right" />
      <div className="Board__edge-snap-maximized" />
    </div>
  );
};
