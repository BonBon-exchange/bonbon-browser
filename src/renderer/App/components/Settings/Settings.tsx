/* eslint-disable jsx-a11y/no-noninteractive-element-interactions */
/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable import/prefer-default-export */
import { useState, ReactElement } from 'react';
import { useTranslation } from 'react-i18next';

import { CloseButton } from 'renderer/App/components/CloseButton';
import { ApplicationSettings } from './ApplicationSettings';
import { BrowsingSettings } from './BrowsingSettings';
import { ExtensionsSettings } from './ExtensionsSettings';

import './style.scss';
import { SettingsProps } from './Types';

export const Settings: React.FC<SettingsProps> = ({
  handleClose,
}: SettingsProps) => {
  const { t } = useTranslation();
  const [selectedView, setSelectedView] = useState<ReactElement>(
    <ApplicationSettings />
  );

  return (
    <div id="Settings__container">
      <CloseButton handleClose={handleClose} />
      <div id="Settings__left-panel">
        <h2>{t('Settings')}</h2>
        <ul>
          <li onClick={() => setSelectedView(<ApplicationSettings />)}>
            {t('Application')}
          </li>
          <li
            onClick={() => setSelectedView(<BrowsingSettings />)}
            data-testid="settings-browsing-link"
          >
            {t('Browsing')}
          </li>
          <li
            onClick={() => setSelectedView(<ExtensionsSettings />)}
            data-testid="settings-extensions-link"
          >
            {t('Extensions')}
          </li>
        </ul>
      </div>
      <div id="Settings__right-panel">{selectedView}</div>
    </div>
  );
};
