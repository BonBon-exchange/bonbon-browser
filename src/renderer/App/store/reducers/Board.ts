/* eslint-disable @typescript-eslint/no-unused-expressions */
import { createSlice, PayloadAction } from '@reduxjs/toolkit';
import { v4 } from 'uuid';

import { WritableDraft } from 'immer/dist/internal';
import { BoardType } from 'renderer/App/components/Board/Types';
import { BrowserProps } from 'renderer/App/components/Browser/Types';

type UpdateBrowserType = {
  browserId: string;
  params: Partial<BrowserProps>;
};

type UpdateBrowserUrlType = {
  url: string;
  browserId: string;
};

type UpdateBrowserLoadingType = {
  isLoading: boolean;
  browserId: string;
};

type UpdateBrowserTitleType = {
  title: string;
  browserId: string;
};

type UpdateBrowserFavType = {
  favicon: string;
  browserId: string;
};

type UpdateBrowserCertificateFingerprintType = {
  certificateErrorFingerprint: string | null;
  browserId: string;
};

interface BoardState {
  board: BoardType;
}

const boardId = v4();
const newBoard = {
  id: boardId,
  label: 'New collection',
  browsers: [],
  activeBrowser: null,
  closedUrls: [],
  isFullSize: false,
  browsersActivity: [],
  height: 0,
};

export const initialState: BoardState = {
  board: newBoard,
};

const addCLosedUrl = (state: WritableDraft<BoardState>, url: string) => {
  if (state.board.closedUrls.length > 20) {
    state.board.closedUrls.splice(0, 1);
  }
  state.board.closedUrls.push(url);
};

const addBrowserActivity = (
  state: WritableDraft<BoardState>,
  browserId: string
) => {
  if (state.board.browsersActivity.length > 20) {
    state.board.browsersActivity.splice(0, 1);
  }
  state.board.browsersActivity = state.board.browsersActivity.filter(
    (b) => b !== browserId
  );
  state.board.browsersActivity.push(browserId);
};

const findAndSelectTabForExtensionsFromBrowserId = (
  state: WritableDraft<BoardState>,
  browserId: string
): void => {
  const wcId = state.board.browsers.find(
    (b) => b.id === browserId
  )?.webContentsId;
  if (wcId) window.app.browser.select(wcId);
};

const setStateActiveBrowser = (
  state: WritableDraft<BoardState>,
  browserId: string
): void => {
  state.board.activeBrowser = browserId;
  addBrowserActivity(state, browserId);
  findAndSelectTabForExtensionsFromBrowserId(state, browserId);
};

const getBrowserIndexFromBrowserId = (
  state: WritableDraft<BoardState>,
  browserId: string
): number => {
  return state.board.browsers.findIndex((b) => b.id === browserId);
};

export const boardSlice = createSlice({
  name: 'board',
  initialState,
  reducers: {
    setBoard: (state, action: PayloadAction<BoardType>) => {
      state.board = action.payload;
      const activeBrowser = state.board.browsers.find(
        (b) => b.id === state.board.activeBrowser
      );
      if (!state.board.activeBrowser || !activeBrowser) {
        if (state.board.browsers.length > 0) {
          setStateActiveBrowser(
            state,
            state.board.browsers[state.board.browsers.length - 1].id
          );
        }
      }
      if (state.board.activeBrowser) {
        findAndSelectTabForExtensionsFromBrowserId(
          state,
          state.board.activeBrowser
        );
      }
    },
    toggleBoardFullSize: (state) => {
      state.board.isFullSize = !state.board.isFullSize;
      window.app.analytics.event(`toggle_fullsize_${!state.board.isFullSize}`);
    },
    setActiveBrowser: (state, action: PayloadAction<string>) => {
      setStateActiveBrowser(state, action.payload);
    },
    addBrowser: (state, action: PayloadAction<BrowserProps>) => {
      if (action.payload.browserIndex) {
        state.board.browsers.splice(
          action.payload.browserIndex,
          0,
          action.payload
        );
      } else {
        state.board.browsers.push(action.payload);
      }
      setStateActiveBrowser(state, action.payload.id);
      window.app.analytics.event('add_browser');
    },
    setBrowsers: (state, action: PayloadAction<BrowserProps[]>) => {
      state.board.browsers = action.payload;
    },
    updateBrowser: (state, action: PayloadAction<UpdateBrowserType>) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex] = {
          ...state.board.browsers[browserIndex],
          ...action.payload.params,
        };
      }
    },
    updateBrowserUrl: (state, action: PayloadAction<UpdateBrowserUrlType>) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].url = action.payload.url;
      }
    },
    updateBrowserLoading: (
      state,
      action: PayloadAction<UpdateBrowserLoadingType>
    ) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].isLoading = action.payload.isLoading;
      }
    },
    updateBrowserTitle: (
      state,
      action: PayloadAction<UpdateBrowserTitleType>
    ) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].title = action.payload.title;
      }
    },
    updateBrowserFav: (state, action: PayloadAction<UpdateBrowserFavType>) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].favicon = action.payload.favicon;
      }
    },
    renameBoard: (state, action: PayloadAction<string>) => {
      state.board.label = action.payload;
      window.app.analytics.event('rename_board');
    },
    minimizeBrowser: (state, action: PayloadAction<string>) => {
      const browserIndex = getBrowserIndexFromBrowserId(state, action.payload);
      state.board.browsers[browserIndex].isMinimized = true;
    },
    unminimizeBrowser: (state, action: PayloadAction<string>) => {
      const browserIndex = getBrowserIndexFromBrowserId(state, action.payload);

      state.board.browsers[browserIndex].isMinimized = false;
    },
    removeBrowser: (state, action: PayloadAction<string>) => {
      const browserIndex = getBrowserIndexFromBrowserId(state, action.payload);

      // remove browser from state
      if (browserIndex > -1) {
        if (!state.board.closedUrls) state.board.closedUrls = [];
        addCLosedUrl(state, state.board.browsers[browserIndex].url);
        state.board.lastClosedBrowserDimensions = [
          state.board.browsers[browserIndex].width,
          state.board.browsers[browserIndex].height,
        ];
        state.board.browsers.splice(browserIndex, 1);

        // clean browsersActivity state
        state.board.browsersActivity = state.board.browsersActivity.filter(
          (b) => b !== action.payload
        );
      }

      // clean activeBrowser
      if (state.board.activeBrowser === action.payload) {
        if (state.board.browsers.length > 0) {
          setStateActiveBrowser(
            state,
            state.board.browsersActivity[
              state.board.browsersActivity.length - 1
            ]
          );
        } else {
          state.board.activeBrowser = null;
          window.app.browser.selectBrowserView();
        }
      }

      window.app.analytics.event('close_browser');
    },
    removeAllBrowsersExcept: (state, action: PayloadAction<string>) => {
      state.board.browsers
        .filter((b) => b.id !== action.payload)
        .forEach((b) => addCLosedUrl(state, b.url));

      state.board.browsers = state.board.browsers.filter(
        (b) => b.id === action.payload
      );

      // clean activeBrowser
      setStateActiveBrowser(state, action.payload);

      // clean browsersActivity state
      state.board.browsersActivity = state.board.browsersActivity.filter(
        (b) => b === action.payload
      );

      // send event
      window.app.analytics.event('close_others_browser');
    },
    updateBrowserCertificateErrorFingerprint: (
      state,
      action: PayloadAction<UpdateBrowserCertificateFingerprintType>
    ) => {
      const browserIndex = getBrowserIndexFromBrowserId(
        state,
        action.payload.browserId
      );
      if (browserIndex > -1) {
        state.board.browsers[browserIndex].certificateErrorFingerprint =
          action.payload.certificateErrorFingerprint;
      }
    },
    removeAllBrowsers: (state) => {
      if (!state.board.closedUrls) state.board.closedUrls = [];
      state.board.browsers.forEach((b) => addCLosedUrl(state, b.url));
      state.board.browsers = [];
      state.board.browsersActivity = [];
      state.board.activeBrowser = null;
      window.app.browser.selectBrowserView();
      window.app.analytics.event('close_all_browsers');
    },
    removeLastClosedUrl: (state) => {
      state.board.closedUrls.splice(state.board.closedUrls.length - 1, 1);
    },
    toggleSearch: (state, action: PayloadAction<string>) => {
      const browserIndex = getBrowserIndexFromBrowserId(state, action.payload);
      state.board.browsers[browserIndex].isSearching =
        !state.board.browsers[browserIndex].isSearching;
    },
    setLastResizedBrowserDimensions: (
      state,
      action: PayloadAction<[number, number]>
    ) => {
      state.board.lastResizedBrowserDimensions = action.payload;
    },
    setBoardHeight: (state, action: PayloadAction<number>) => {
      state.board.height = action.payload;
    },
  },
});

export const {
  setBoard,
  updateBrowserUrl,
  updateBrowserFav,
  renameBoard,
  removeBrowser,
  minimizeBrowser,
  unminimizeBrowser,
  removeAllBrowsers,
  addBrowser,
  setActiveBrowser,
  updateBrowserTitle,
  updateBrowser,
  toggleBoardFullSize,
  removeLastClosedUrl,
  setBrowsers,
  removeAllBrowsersExcept,
  updateBrowserLoading,
  updateBrowserCertificateErrorFingerprint,
  toggleSearch,
  setLastResizedBrowserDimensions,
  setBoardHeight,
} = boardSlice.actions;

export default boardSlice.reducer;
