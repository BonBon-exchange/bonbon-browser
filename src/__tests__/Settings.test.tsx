/* eslint-disable react/jsx-props-no-spreading */
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { Middleware } from '@reduxjs/toolkit';

import { Settings } from '../renderer/App/components/Settings';
import { mockWindow } from './beforeAll';
import { initialState } from '../renderer/App/store/reducers/Board';

let store: any;
const middlewares: Middleware[] = [];

const props = {
  handleClose: jest.fn(),
};

describe('Settings', () => {
  beforeAll(() => {
    mockWindow();
    const mockStore = configureStore(middlewares);
    store = mockStore({ board: initialState });
  });

  it('should render', () => {
    expect(
      render(
        <Provider store={store}>
          <Settings {...props} />
        </Provider>
      )
    ).toBeTruthy();
  });
});
