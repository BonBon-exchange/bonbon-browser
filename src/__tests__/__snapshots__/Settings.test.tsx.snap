// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings should render 1`] = `
<div
  id="Settings__container"
>
  <div
    className="CloseButton__container"
    data-testid="close-button"
    onClick={[MockFunction]}
  >
    <svg
      aria-hidden={true}
      className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="CloseIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
      />
    </svg>
  </div>
  <div
    id="Settings__left-panel"
  >
    <h2>
      Settings
    </h2>
    <ul>
      <li
        onClick={[Function]}
      >
        Application
      </li>
      <li
        data-testid="settings-browsing-link"
        onClick={[Function]}
      >
        Browsing
      </li>
      <li
        data-testid="settings-extensions-link"
        onClick={[Function]}
      >
        Extensions
      </li>
    </ul>
  </div>
  <div
    id="Settings__right-panel"
  >
    <h2>
      Application
    </h2>
    <div
      className="Settings__item"
    >
      <input
        checked={false}
        id="application-settings-launch-at-startup"
        onChange={[Function]}
        type="checkbox"
      />
      <label
        htmlFor="application-settings-launch-at-startup"
      >
        Launch at startup
      </label>
      <div
        className="Settings__item-description"
      >
        If checked, the app will auto-launch at system startup.
      </div>
    </div>
    <div
      className="Settings__item"
    >
      <label
        htmlFor="application-settings-language"
      >
        Language
        :
      </label>
      <select
        id="application-settings-language"
        onChange={[Function]}
      >
        <option
          value="en"
        >
          English
        </option>
        <option
          value="es"
        >
          Spanish
        </option>
        <option
          value="fr"
        >
          French
        </option>
        <option
          value="de"
        >
          German
        </option>
        <option
          value="ru"
        >
          Russian
        </option>
        <option
          value="fa"
        >
          Persian
        </option>
        <option
          value="tr"
        >
          Turkish
        </option>
        <option
          value="cn"
        >
          Chinese
        </option>
        <option
          value="ar"
        >
          Arabic
        </option>
        <option
          value="nl"
        >
          Dutch
        </option>
        <option
          value="pl"
        >
          Polish
        </option>
        <option
          value="pt"
        >
          Portuguese
        </option>
        <option
          value="ja"
        >
          Japanese
        </option>
      </select>
      <div
        className="Settings__item-description"
      >
        The language in which displays the application.
      </div>
    </div>
  </div>
</div>
`;

exports[`Settings should show Browsing 1`] = `
"<div id=\\"Settings__container\\">
  <div class=\\"CloseButton__container\\" data-testid=\\"close-button\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CloseIcon\\">
      <path d=\\"M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path>
    </svg></div>
  <div id=\\"Settings__left-panel\\">
    <h2>Settings</h2>
    <ul>
      <li>Application</li>
      <li data-testid=\\"settings-browsing-link\\">Browsing</li>
      <li data-testid=\\"settings-extensions-link\\">Extensions</li>
    </ul>
  </div>
  <div id=\\"Settings__right-panel\\">
    <h2>Browsing</h2>
    <div class=\\"Settings__item\\" data-testid=\\"settings-browsing-page\\"><label for=\\"browsing-settings-default-webpage\\">New windows default webpage:</label><input type=\\"text\\" id=\\"browsing-settings-default-webpage\\" value=\\"\\">
      <div class=\\"Settings__item-description\\">The url that will open first when opening a new window.</div>
    </div>
    <div class=\\"Settings__item\\"><label for=\\"browsing-settings-default-search-engine\\">Default search engine:</label><select id=\\"browsing-settings-default-search-engine\\">
        <option value=\\"google\\">Google</option>
        <option value=\\"presearch\\">Presearch</option>
        <option value=\\"qwant\\">Qwant</option>
        <option value=\\"duckduckgo\\">DuckDuckGo</option>
        <option value=\\"yandex\\">Yandex</option>
        <option value=\\"swisscows\\">Swisscows</option>
        <option value=\\"ecosia\\">Ecosia</option>
        <option value=\\"startpage\\">StartPage</option>
      </select>
      <div class=\\"Settings__item-description\\">The search engine that will be used when entering text in the url input and pressing enter.</div>
    </div>
    <div class=\\"Settings__item\\">
      <div class=\\"Settings__item-title\\">New webpages default size</div><input type=\\"radio\\" name=\\"browsing-settings-default-size\\" id=\\"browsing-settings-default-size-defined\\" value=\\"defined\\"><label for=\\"browsing-settings-default-size-defined\\">As defined here:</label><input type=\\"number\\" id=\\"browsing-settings-default-width\\" value=\\"0\\">x<input type=\\"number\\" id=\\"browsing-settings-default-height\\" value=\\"0\\"><br><input type=\\"radio\\" name=\\"browsing-settings-default-size\\" id=\\"browsing-settings-default-size-last-closed\\" value=\\"lastClosed\\"><label for=\\"browsing-settings-default-size-last-closed\\">Use the size of the last closed webpage</label><br><input type=\\"radio\\" name=\\"browsing-settings-default-size\\" id=\\"browsing-settings-default-size-last-resized\\" value=\\"lastResized\\"><label for=\\"browsing-settings-default-size-last-resized\\">Use the size of the last resized window</label>
      <div class=\\"Settings__item-description\\">The size of opening new windows when they are not maximized.</div>
    </div>
    <div class=\\"Settings__item\\"><input type=\\"checkbox\\" id=\\"browsing-settings-dont-save-history\\"><label for=\\"browsing-settings-dont-save-history\\">Do not save browsing history</label>
      <div class=\\"Settings__item-description\\">If checked, BonBon will not save your browsing history. It will also disable URL suggestions.</div>
    </div>
    <div class=\\"Settings__item\\">
      <div class=\\"Settings__item-title\\">When dragging a window to the top edge:</div><input type=\\"radio\\" name=\\"browsing-settings-top-edge\\" id=\\"browsing-settings-top-edge-maximize\\" value=\\"maximize\\"><label for=\\"browsing-settings-top-edge-maximize\\">Maximize</label><br><input type=\\"radio\\" name=\\"browsing-settings-top-edge\\" id=\\"browsing-settings-top-edge-fit\\" value=\\"fit\\"><label for=\\"browsing-settings-top-edge-fit\\">Fit the screen</label>
      <div class=\\"Settings__item-description\\">Maximize the window or enlarge the window when dragging it to the top edge.</div>
    </div>
  </div>
</div>"
`;

exports[`Settings should show Extensions 1`] = `
"<div id=\\"Settings__container\\">
  <div class=\\"CloseButton__container\\" data-testid=\\"close-button\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CloseIcon\\">
      <path d=\\"M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path>
    </svg></div>
  <div id=\\"Settings__left-panel\\">
    <h2>Settings</h2>
    <ul>
      <li>Application</li>
      <li data-testid=\\"settings-browsing-link\\">Browsing</li>
      <li data-testid=\\"settings-extensions-link\\">Extensions</li>
    </ul>
  </div>
  <div id=\\"Settings__right-panel\\">
    <h2>Extensions</h2>
    <div class=\\"Settings__item\\" data-testid=\\"settings-extensions-page\\"><input type=\\"checkbox\\" id=\\"extensions-force-install-u-block\\"><label for=\\"extensions-force-install-u-block\\">Force uBlockOrigin installation at startup</label>
      <div class=\\"Settings__item-description\\">If checked, the app will automatically download uBlockOrigin and install it when starting.</div>
    </div>
  </div>
</div>"
`;

exports[`Settings should show first screen 1`] = `
"<div id=\\"Settings__container\\">
  <div class=\\"CloseButton__container\\" data-testid=\\"close-button\\"><svg class=\\"MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root\\" focusable=\\"false\\" aria-hidden=\\"true\\" viewBox=\\"0 0 24 24\\" data-testid=\\"CloseIcon\\">
      <path d=\\"M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z\\"></path>
    </svg></div>
  <div id=\\"Settings__left-panel\\">
    <h2>Settings</h2>
    <ul>
      <li>Application</li>
      <li data-testid=\\"settings-browsing-link\\">Browsing</li>
      <li data-testid=\\"settings-extensions-link\\">Extensions</li>
    </ul>
  </div>
  <div id=\\"Settings__right-panel\\">
    <h2>Application</h2>
    <div class=\\"Settings__item\\"><input type=\\"checkbox\\" id=\\"application-settings-launch-at-startup\\"><label for=\\"application-settings-launch-at-startup\\">Launch at startup</label>
      <div class=\\"Settings__item-description\\">If checked, the app will auto-launch at system startup.</div>
    </div>
    <div class=\\"Settings__item\\"><label for=\\"application-settings-language\\">Language:</label><select id=\\"application-settings-language\\">
        <option value=\\"en\\">English</option>
        <option value=\\"es\\">Spanish</option>
        <option value=\\"fr\\">French</option>
        <option value=\\"de\\">German</option>
        <option value=\\"ru\\">Russian</option>
        <option value=\\"fa\\">Persian</option>
        <option value=\\"tr\\">Turkish</option>
        <option value=\\"cn\\">Chinese</option>
        <option value=\\"ar\\">Arabic</option>
        <option value=\\"nl\\">Dutch</option>
        <option value=\\"pl\\">Polish</option>
        <option value=\\"pt\\">Portuguese</option>
        <option value=\\"ja\\">Japanese</option>
      </select>
      <div class=\\"Settings__item-description\\">The language in which displays the application.</div>
    </div>
  </div>
</div>"
`;
